linetools/analysis/continuumfnd.py

```diff
@@ -0,0 +1,206 @@
+      output file with knots (.jsn or .json type)
+    overwrite : bool
+      overwrite?
+
+    Returns:
+    --------
+    knots : list of 2-element lists
+      List of knots (wavelength, flux) that define continuum.
+    knotpixs : list of int
+      Indices of pixels that define continuum
+
+    """
+
+    # input spectrum
+    if type(spec) is xspec.XSpectrum1D:
+        sp = spec
+    else:
+        try:
+            sp = readspec(spec)
+        except OSError:
+            print('Not valid input. Please provide XSpectrum1D object, or a file that contains XSpectrum1D object')
+            return
+
+    wav = sp.wavelength
+    flx = sp.flux
+    sigf = sp.sig
+
+    # smoothed input spectrum by smsm pixels
+    sp3 = sp.box_smooth(sm)
+    wav3 = sp3.wavelength
+    flx3 = sp3.flux
+    sigf3 = sp3.sig
+
+
+    # 1) Define pixels that are on the continuum.
+    #    Smooth the spectrum, find pixels around maximums in the flux in the smoothed spectrum,
+    #    assume that the continuum is well defined by these pixels only.
+    #    This flux will be too high, and is used only in the step (1)
+    ipixs = []  # continuum pixels
+    for i in range(len(flx3)):
+        i1 = np.max([0, i - npix])
+        i2 = np.min([i + npix, (len(flx3)) - 1])
+        ii = np.linspace(i1, i2, i2 - i1 + 1)
+        isum = 0
+        for j in ii:
+            if (flx3[int(j)] <= flx3[i]):
+                isum = isum + 1
+            if (flx3[int(j)] >= flx3[i]):
+                isum = isum - 1
+        if (isum) == len(ii) - 1:
+            ipixs.append(i)
+
+    # interpolate continuum to all pixels
+    cont0ipixs = flx3[ipixs]
+    cont0 = np.interp(wav, wav3[ipixs], cont0ipixs)
+
+
+    # 2) Identify pixels that correspond to absorption lines. Identify all other pixels.
+    #    Define continuum as in (1),
+    #    define absorption line as a set of adjacent pixels that are below the continuum level; flux of the boundary
+    #      pixels exceeds continuum level
+    #    Define EW = sum of (1 - flux/continuum) of these adjacent pixels
+    #    and a condition that EW > ewsnlim * sigma(EW) ; default ewsnlim = 2
+
+    # find pixels that correspond to absorption lines
+    ind0s = [] # pixels corresponding to absorption lines
+    imin = -1
+    iew = 0
+    isigew = 0
+
+    for i in range(len(flx)):
+        if flx[i] == 0:
+            ind0s.append(i)
+        else:
+            if flx[i] < cont0[i]:
+                iew = iew + (1 - flx[i] / cont0[i])
+                isigew = (isigew ** 2 + (sigf[i] / cont0[i]) ** 2) ** 0.5
+                if imin == -1:
+                    imin = np.max([i - 1, 0])
+            else:
+                if (iew > ewsnlim * isigew):
+                    for j in range(i - imin + 1):
+                        ind0s.append(j + imin)
+                iew = 0
+                isigew = 0
+                imin = -1
+
+    # Find pixels that do not correspond to absorption lines
+    ind0s = list(set(ind0s))
```

Comment:
Why `list` manipulation.  Just use array + indices.

Reply:
Addressed with the previous change.